lib/MahoCLI/Commands/SysEncryptionKeyRegenerate.php

```diff
@@ -0,0 +1,279 @@
+
+        // Write the updated configuration back to the file
+        if (file_put_contents($localXmlPath, $updatedContent) === false) {
+            $output->writeln('<error>Failed to write updated configuration</error>');
+            return Command::FAILURE;
+        }
+
+        $output->writeln('<info>Encryption key has been successfully updated</info>');
+        $output->writeln('<comment>New key: ' . $newKey . '</comment>');
+        $output->writeln('');
+
+        $readConnection = Mage::getSingleton('core/resource')->getConnection('core_read');
+        $writeConnection = Mage::getSingleton('core/resource')->getConnection('core_write');
+
+        $this->recryptAdminUserTable($output, $readConnection, $writeConnection);
+        $this->recryptCoreConfigDataTable($output, $readConnection, $writeConnection);
+
+        Mage::dispatchEvent('encryption_key_regenerated', [
+            'output' => $output,
+            'encrypt_callback' => [$this, 'encrypt'],
+            'decrypt_callback' => [$this, 'decrypt'],
+        ]);
+
+        if (\Composer\InstalledVersions::isInstalled('mahocommerce/module-mcrypt-compat')) {
+            $output->writeln('');
+            $output->writeln('<error>You may now remove the compatibility module using: composer remove mahocommerce/module-mcrypt-compat</error>');
+            $output->writeln('Then check if you still have phpseclib/mcrypt_compat installed and evaluate its removal too.');
+            $output->writeln('');
+        } elseif (\Composer\InstalledVersions::isInstalled('phpseclib/mcrypt_compat')) {
+            $output->writeln('');
+            $output->writeln('<error>Warning: phpseclib/mcrypt_compat is installed. This package is obsolete and should be removed.</error>');
+            $output->writeln('If directly installed, remove it with <info>composer remove phpseclib/mcrypt_compat</info>.');
+            $output->writeln('If installed as a dependency, find which package requires it with <info>composer why phpseclib/mcrypt_compat</info> and evaluate its removal.');
+            $output->writeln('');
+        }
+
+        return Command::SUCCESS;
+    }
+
+    protected function recryptAdminUserTable(OutputInterface $output, \Varien_Db_Adapter_Interface $readConnection, \Varien_Db_Adapter_Interface $writeConnection): void
+    {
+        $output->write('Re-encrypting data on admin_user table... ');
+
+        $table = Mage::getSingleton('core/resource')->getTableName('admin_user');
+        $select = $readConnection->select()
+            ->from($table)
+            ->where('twofa_secret IS NOT NULL');
+        $encryptedData = $readConnection->fetchAll($select);
+
+        foreach ($encryptedData as $encryptedDataRow) {
+            $writeConnection->update(
+                $table,
+                ['twofa_secret' => $this->encrypt($this->decrypt($encryptedDataRow['twofa_secret']))],
+                ['user_id = ?' => $encryptedDataRow['user_id']],
+            );
+        }
+
```

Comment:
```suggestion

        Mage::app()->getCacheInstance()->clean('config');

```

Reply:
I eneded up doing it in a slightly different way

<img width="661" alt="Screenshot 2025-03-31 alle 18 43 51" src="https://github.com/user-attachments/assets/95d806ee-269e-4b3b-832f-633151cb4468" />

one before the event (so that custom code reads everything as updated as possible) and then after the custom observers